aware-core/src/main/java/com/aware/Bluetooth.java

```diff
@@ -163,14 +171,28 @@ public int onStartCommand(Intent intent, int flags, int startId) {
             } else {
                 save_bluetooth_device(bluetoothAdapter);
 
-                if (FREQUENCY != Integer.parseInt(Aware.getSetting(getApplicationContext(), Aware_Preferences.FREQUENCY_BLUETOOTH))) {
-                    alarmManager.cancel(bluetoothScan);
-                    alarmManager.setRepeating(AlarmManager.RTC_WAKEUP,
-                            System.currentTimeMillis() + 1000,
-                            Integer.parseInt(Aware.getSetting(getApplicationContext(), Aware_Preferences.FREQUENCY_BLUETOOTH)) * 1000,
-                            bluetoothScan);
+                int NEW_FREQUENCY = Integer.parseInt(Aware.getSetting(getApplicationContext(), Aware_Preferences.FREQUENCY_BLUETOOTH));
+                if (FREQUENCY != NEW_FREQUENCY) {
+
+                    Scheduler.Schedule schedule = Scheduler.getSchedule(this, SCHEDULE_BLUETOOTH);
+
+                        // sync is null, or sync is not null but an interval is detected or if no hours AND minutes are detected, recreate the schedule
+                    try {
+                        if (schedule == null || (schedule != null && schedule.getHours().length() > 0 && schedule.getHours().getInt(0) != NEW_FREQUENCY)) {
```

Comment:
schedule.getInterval() should give you what you need. Why are we considering hours and minutes here?